Jobsway2goMvc/Controllers/CollectionsController.cs

```diff
@@ -0,0 +1,179 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.EntityFrameworkCore;
+using Jobsway2goMvc.Data;
+using Jobsway2goMvc.Models;
+using Microsoft.AspNetCore.Identity;
+using System.Security.Claims;
+using Microsoft.Extensions.Hosting;
+
+namespace Jobsway2goMvc.Controllers
+{
+    public class CollectionsController : Controller
+    {
+        private readonly ApplicationDbContext _context;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public CollectionsController(ApplicationDbContext context,IHttpContextAccessor httpContextAccessor)
+        {
+            _context = context;
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        // GET: Collections
+        public async Task<IActionResult> Index()
+        {
+              return View(await _context.Collections.ToListAsync());
+        }
+
+        // GET: Collections/Details/5
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null || _context.Collections == null)
+            {
+                return NotFound();
+            }
+
+            var collection = await _context.Collections
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (collection == null)
+            {
+                return NotFound();
+            }
+
+            return View(collection);
+        }
+
+        // GET: Collections/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        private ApplicationUser GetUser(ClaimsPrincipal principal)
+        {
+            var userId = principal.FindFirstValue(ClaimTypes.NameIdentifier);
+
+            var user = _context.Users.FirstOrDefault(u => u.Id == userId);
+
+            return user;
+        }
+
+        // POST: Collections/Create
+        // To protect from overposting attacks, enable the specific properties you want to bind to.
+        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("Id,Name")] Collection collection)
+        {
+            ModelState.Remove("User");
+            if (ModelState.IsValid)
+            {
+                var user = _httpContextAccessor.HttpContext.User;
+                var applicationUser = GetUser(user);
+                collection.User = applicationUser;
+                _context.Add(collection);
```

Comment:
Remove `var applicationUser = GetUser(user);` and make it directly `collection.User = GetUser(user);`